src/libraries/System.Collections/src/System/Collections/Generic/SortedDictionary.cs

```diff
@@ -938,6 +948,14 @@ public override int Compare(KeyValuePair<TKey, TValue> x, KeyValuePair<TKey, TVa
             {
                 return keyComparer.Compare(x.Key, y.Key);
             }
+
+            public override bool Equals(object? obj)
```

Comment:
Couple of issues with this change:

1. This changes the equality semantics of a public class, which is technically a breaking change. While `KeyValuePairComparer` is probably not a very commonly used type and I would not expect existing users to rely on its current behaviour. Still, I don't trust that the risk is worth the benefit of that particular optimization.
2. It assumes that that nested comparer type is a singleton, which might be true for `Comparer<T>.Default` but probably not for user supplied comparers.

@stephentoub  thoughts?

Reply:
It seems while the comparer class is declared public, it's actually missing from the reference assemblies. While the code shared [here](https://github.com/dotnet/runtime/pull/45659/files#r537329890) is valid in C# interactive, it fails to compile in regular in C# projects.

Reply:
> It seems while the comparer class is declared public, it's actually missing from the reference assemblies.

It is needed to make binary serialization compatible with desktop.

Reply:
From that perspective then, we probably wouldn't care much if equality were changed?